src/System.Diagnostics.Debug/tests/DebugTests.cs

```diff
@@ -200,15 +342,12 @@ static void VerifyAssert(Action test, params string[] expectedOutputStrings)
                 return;
             }
 
-            FieldInfo writeCoreHook = typeof(Debug).GetField("s_WriteCore", BindingFlags.Static | BindingFlags.NonPublic);
-            FieldInfo showDialogHook = typeof(Debug).GetField("s_ShowDialog", BindingFlags.Static | BindingFlags.NonPublic);
+            FieldInfo writeCoreHook = typeof(DebugProvider).GetField("s_WriteCore", BindingFlags.Static | BindingFlags.NonPublic);
+            Debug.SetProvider(WriteLogger.s_instance);
```

Comment:
The `SetProvider` return value is never used anywhere. Do you plan to ever use it? If not, it may be better to change it to return `void`.

Reply:
Looking....